pkg/alertmanager/state_replication.go

```diff
@@ -167,7 +173,22 @@ func (s *state) starting(ctx context.Context) error {
 		}
 	}
 
-	level.Info(s.logger).Log("msg", "state not settled but continuing anyway", "err", err)
+	level.Info(s.logger).Log("msg", "state not settled; trying to read from storage", "err", err)
+
+	// Attempt to read the state from persistent storage instead.
+	storeReadCtx, cancel := context.WithTimeout(ctx, s.storeReadTimeout)
+	defer cancel()
+
+	fullState, err := s.store.GetFullState(storeReadCtx, s.userID)
+	if err == nil {
```

Comment:
To improve the quality of logs, I would check if the error is a "object not found". If so, I would log that no state is in the store (think about a new tenant) and then return.